indico/modules/events/timetable/client/js/actions.ts

```diff
@@ -304,3 +302,59 @@ export function closeModal() {
 export function setCurrentDate(date: Moment, eventId: number) {
   return {type: SET_CURRENT_DATE, date, eventId};
 }
+
+/**
+ * Manages a queue of requests to be processed in order.
+ * This ensures all timetable requests are handled sequentially.
+ *
+ * TODO(tomas): On error, we just call `handleAxiosError` for now.
+ * In the future, we might consider some undo mechanism.
+ */
+class RequestQueue {
+  requests: (() => Promise<object>)[];
+  working: boolean;
+
+  constructor() {
+    this.requests = [];
+    this.working = false;
+  }
```

Comment:
Can't we just put these values as default? We're not passing anything through the constructor so maybe at attribute level we can assign default values,

So for example `working: boolean = false;`. I think if the point is to have some typing, then here maybe it's not so useful because these values have very obvious types? Just an idea